sdk/maps/azure-maps-search/azure/maps/search/_search_client.py

```diff
@@ -65,1050 +45,387 @@ def __init__(
-            position of the bounding box as float. E.g. BoundingBox(west=37.553, south=-122.453, east=33.2, north=57)
-        :keyword str language:
-            Language in which search results should be returned.
-        :keyword extended_postal_codes_for:
-            Indexes for which extended postal codes should be included in the results.
-        :paramtype extended_postal_codes_for:
-            list[str or ~azure.maps.search.models.SearchIndexes]
-        :keyword list[str] brand_filter:
-            A comma-separated list of brand names which could be used to restrict the
-            result to specific brands. Item order does not matter.
-        :keyword electric_vehicle_connector_filter:
-            A comma-separated list of connector types which could
-            be used to restrict the result to Electric Vehicle Station supporting specific connector types.
-        :paramtype electric_vehicle_connector_filter:
-            list[str or ~azure.maps.search.models.ElectricVehicleConnector]
-        :keyword localized_map_view:
-            The View parameter (also called the "user region" parameter) allows
-            you to show the correct maps for a certain country/region for geopolitically disputed regions.
-        :paramtype localized_map_view:
-            str or ~azure.maps.search.models.LocalizedMapView
-        :keyword operating_hours:
-            Hours of operation for a POI (Points of Interest). The availability of
-            hours of operation will vary based on the data available. If not passed, then no opening hours
-            information will be returned.
-            Supported value: nextSevenDays.
-        :paramtype operating_hours:
-            str or ~azure.maps.search.models.OperatingHoursRange
-        :return:
-            The results of search.
-        :rtype:
-            ~azure.maps.search.models.SearchAddressResult
-        :raises:
-            ~azure.core.exceptions.HttpResponseError
-        """
-
-        coordinates = kwargs.pop("coordinates", (0, 0))
-        bounding_box = kwargs.pop("bounding_box", BoundingBox())
-
-        result = self._search_client.search_point_of_interest_category(
-            query,
-            lat=coordinates[0],
-            lon=coordinates[1],
-            btm_right=f"{bounding_box.south}, {bounding_box.east}",
-            top_left=f"{bounding_box.north}, {bounding_box.west}",
-            **kwargs
-        )
-        return SearchAddressResult(result.summary, result.results)
-
-    @distributed_trace
-    def search_address(
-        self,
-        query: str,
-        **kwargs: Any
-    ) -> SearchAddressResult:
-        """**Address Geocoding**
+    def get_geocoding(
+            self,
+            *,
+            top: int = 5,
+            query: Optional[str] = None,
+            address_line: Optional[str] = None,
+            country_region: Optional[str] = None,
+            bbox: Optional[List[float]] = None,
```

Comment:
We should rename it as bounding_box and usinging type `BoundingBox` (self-defined)